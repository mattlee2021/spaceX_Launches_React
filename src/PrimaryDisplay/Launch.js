import React from "react";
import ReactPlayer from "react-player";
import NotesBox from "../Features/Notesbox";
import styles from "./Launch.module.css";

const Launch = ({ launch }) => {
  const launchUTCTime = new Date(launch.launch_date_utc);
  const launchLocalTime = launchUTCTime.toLocaleString();
  const localTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
  return (
    <div className={styles.launch__items}>
      <p>Mission Name: {launch.mission_name}</p>
      <p>Launch Date: {launchLocalTime + " " + localTimeZone}</p>
      <p>Launch Site: {launch.launch_site.site_name_long}</p>
      <p>Rocket Name: {launch.rocket.rocket_name}</p>

      <ReactPlayer
        url={launch.links.video_link}
        controls
        className={styles.launch__video}
      />
      <NotesBox />
      <hr className={styles.line__solid} />
    </div>
  );
};

export default Launch;
