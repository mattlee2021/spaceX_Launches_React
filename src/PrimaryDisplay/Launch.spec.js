import TestRenderer from "react-test-renderer";
import Launch from "./Launch";
describe("Launch", () => {
  const mockLaunchData = {
    launch_data_utc: "2020-10-24T15:31:00.000Z",
    mission_name: "Starlink-15 (v1.0)",
    launch_site: {
      site_name_long:
        "Cape Canaveral Air Force Station Space Launch Complex 40",
    },
    rocket: {
      rocket_name: "Falcon 9",
    },
    links: {
      video_link: "mockVideoLink.com",
    },
  };
  it("Should match the snapshot with the mocked data", () => {
    const testRenderer = TestRenderer.create(
      <Launch launch={mockLaunchData} />
    );
    testRenderer.toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});
