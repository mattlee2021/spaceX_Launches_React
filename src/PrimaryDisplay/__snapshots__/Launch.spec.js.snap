// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Should match the snapshot with the mocked data 1`] = `
<div
  className="launch__items"
>
  <p>
    Mission Name: 
    Starlink-15 (v1.0)
  </p>
  <p>
    Launch Date: 
    Invalid Date America/New_York
  </p>
  <p>
    Launch Site: 
    Cape Canaveral Air Force Station Space Launch Complex 40
  </p>
  <p>
    Rocket Name: 
    Falcon 9
  </p>
  <div
    className="launch__video"
    style={
      Object {
        "height": "360px",
        "width": "640px",
      }
    }
  />
  <form
    className="notes_form"
  >
    <div>
      <label>
        Notes
      </label>
      <br />
      <br />
      <textarea
        className="notes__input"
        onChange={[Function]}
        value=""
      />
    </div>
    <br />
    <button
      onClick={[Function]}
      type="button"
    >
      Copy This Note To Clipboard
    </button>
  </form>
  <hr
    className="line__solid"
  />
</div>
`;
